java/code/src/com/redhat/rhn/common/messaging/SmtpMail.java

```diff
@@ -65,16 +65,37 @@ public SmtpMail() {
 
         Config c = Config.get();
         smtpHost = c.getString(ConfigDefaults.WEB_SMTP_SERVER, "localhost");
+        Integer smtpPort = c.getInt(ConfigDefaults.WEB_SMTP_PORT, 25);
+        Boolean smtpAuth = c.getBoolean(ConfigDefaults.WEB_SMTP_AUTH);
+        Boolean smtpSSL = c.getBoolean(ConfigDefaults.WEB_SMTP_SSL);
+        Boolean smtpStartTLS = c.getBoolean(ConfigDefaults.WEB_SMTP_STARTTLS);
+        String smtpUser = c.getString(ConfigDefaults.WEB_SMTP_USER);
+        String smtpPass = c.getString(ConfigDefaults.WEB_SMTP_PASS);
         String from = c.getString(ConfigDefaults.WEB_DEFAULT_MAIL_FROM, "root@localhost");
 
         // Get system properties
         Properties props = System.getProperties();
 
         // Setup mail server
         props.put("mail.smtp.host", smtpHost);
+        props.put("mail.smtp.port", smtpPort);
+        props.put("mail.smtp.auth", smtpAuth);
+        props.put("mail.smtp.ssl.enable", smtpSSL);
+        props.put("mail.smtp.starttls.enable", smtpStartTLS);
+        props.put("mail.smtp.ssl.protocols", "TLSv1.2");
```

Comment:
Can we give multiple values here? I think TLSv1.3 becomes more popular these days.